File: src/main/动态规划/t746/minCostClimbingStairs.go
package main

func minCostClimbingStairs(cost []int) int {
	min := func(a, b int) int {
		if a > b {
			return b
		}
		return a
	}
	n := len(cost)
	for i := 2; i < n; i++ {
		cost[i] += min(cost[i-1], cost[i-2])
	}
	return min(cost[n-1], cost[n-2])
}

File: src/main/动态规划/t746/MinCostClimbingStairs.java
package 动态规划.t746;

/**
 * @author: sotowang
 * @date: 2020/12/21 15:27
 */
public class MinCostClimbingStairs {
    public int minCostClimbingStairs(int[] cost) {
        for(int i=2;i<cost.length;i++){
            cost[i] += Math.min(cost[i-2],cost[i-1]);
        }
        return Math.min(cost[cost.length-1],cost[cost.length-2]);
    }
}
